code/modules/mob/living/carbon/xenomorph/abilities/crusher/crusher_powers.dm

```diff
@@ -164,3 +215,78 @@
 		to_chat(X, SPAN_XENOHIGHDANGER("You feel your enhanced shield end!"))
 
 	X.overlay_shields()
+
+/datum/action/xeno_action/onclick/charger_charge/use_ability(atom/A)
+	var/mob/living/carbon/Xenomorph/X = owner
+
+	activated = !activated
+	var/will_charge = "[activated ? "now" : "no longer"]"
+	to_chat(X, SPAN_XENONOTICE("You will [will_charge] charge when moving."))
+	if(activated)
+		RegisterSignal(X, COMSIG_MOVABLE_MOVED, .proc/handle_movement)
+		RegisterSignal(X, COMSIG_ATOM_DIR_CHANGE, .proc/handle_dir_change)
+		RegisterSignal(X, COMSIG_XENO_RECALCULATE_SPEED, .proc/update_speed)
+		RegisterSignal(X, COMSIG_XENO_STOP_MOMENTUM, .proc/stop_momentum)
+		RegisterSignal(X, COMSIG_MOVABLE_ENTERED_RIVER, .proc/handle_river)
+		RegisterSignal(X, COMSIG_LIVING_PRE_COLLIDE, .proc/handle_collision)
+		RegisterSignal(X, COMSIG_XENO_START_CHARGING, .proc/start_charging)
+		button.icon_state = "template_on"
+	else
+		stop_momentum()
+		UnregisterSignal(X, list(
+			COMSIG_MOVABLE_MOVED,
+			COMSIG_ATOM_DIR_CHANGE,
+			COMSIG_XENO_RECALCULATE_SPEED,
+			COMSIG_MOVABLE_ENTERED_RIVER,
+			COMSIG_LIVING_PRE_COLLIDE,
+			COMSIG_XENO_STOP_MOMENTUM,
+			COMSIG_XENO_START_CHARGING,
+			button.icon_state = "template"
+		))
+	if(!activated)
+		button.icon_state = "template"
+
+/datum/action/xeno_action/activable/croosh/use_ability(atom/A)
+	var/mob/living/carbon/Xenomorph/X = owner
+
+/datum/action/xeno_action/activable/tumble/use_ability(atom/A)
+	if(!action_cooldown_check())
```

Comment:
Same thing with these single letter variable names. What is A supposed to be, the target?